Library/Homebrew/formula_auditor.rb

```diff
@@ -900,6 +902,81 @@ def audit_revision
             current_revision > (newest_committed[:revision] + 1)
         problem "`revision` should only increment by 1"
       end
+
+      revision_increment = if (previous_revision = previous_committed[:revision])
+        current_revision - previous_revision
+      else
+        current_revision
+      end
+      return if revision_increment != 1
+
+      dependency_names = formula.recursive_dependencies.map(&:name)
+      return if dependency_names.empty?
+
+      changed_dependency_paths = changed_formulae_paths(tap, only_names: dependency_names)
+      return if changed_dependency_paths.empty?
+
+      missing_compatibility_bumps = changed_dependency_paths.filter_map do |path|
+        changed_formula = Formulary.factory(path)
+        # Each changed dependency must raise its compatibility_version by exactly one.
+        previous_committed_dependency, = committed_version_info(formula: changed_formula)
+        previous_compatibility_version = previous_committed_dependency[:compatibility_version] || 0
+        current_compatibility_version = changed_formula.compatibility_version || 0
+        next if current_compatibility_version == previous_compatibility_version + 1
+
+        "#{changed_formula.name} (#{previous_compatibility_version} to #{current_compatibility_version})"
+      end
+      return if missing_compatibility_bumps.empty?
+
+      problem "`revision` increased but recursive dependencies must increase `compatibility_version` by 1: " \
+              "#{missing_compatibility_bumps.join(", ")}."
+    end
+
+    def audit_compatibility_version
+      return unless @git
+
+      tap = formula.tap
+      return if tap.nil?
+      return unless tap.git?
+
+      previous_committed, = committed_version_info
```

Comment:
Same here I think:

```suggestion
      _, newest_committed = committed_version_info
```

We are again comparing relative to `origin/HEAD`